test/unit/test_checkpointing.py

```diff
@@ -0,0 +1,230 @@
+import os
+import shutil
+import pathlib
+import tempfile
+import time
+import unittest
+from unittest.mock import patch
+import numpy as np
+import xgboost as xgb
+from sagemaker_xgboost_container import checkpointing
+from sagemaker_xgboost_container.checkpointing import SaveCheckpoint
+
+
+class TestSaveCheckpoint(unittest.TestCase):
+
+    def setUp(self):
+
```

Comment:
Super nitpick and personal choice, but you have a lot of unneeded new lines in this file :)

Reply:
On retrying, the thought was there's a chance that some files are intact even if others are corrupt. We retain N=5 checkpoints (N retires is basically why we save N checkpoints), and I think it's much cheaper for users to save N checkpoints and try them all than to train from scratch if 1 fails.

Ran `python3 -m tox` and confirmed that all tests/flake8 pass.